python/nano/src/bigdl/nano/__init__.py

```diff
@@ -41,6 +43,47 @@ def _check_nano_envs():
 # _check_nano_envs()
 
 
+@lru_cache(maxsize=None)
```

Comment:
why adding a cache here? Will `get_patch_map` be called many times?

Reply:
It doesn't matter if we only need `patch_nano`. This cache here is to make this function return the same reference every time. This is really important if I want the "original" class back during `unpatch_nano`.

Inspired by https://github.com/intel/scikit-learn-intelex/blob/develop/daal4py/sklearn/monkeypatch/dispatcher.py#L57

Reply:
Does the following code work? Seems more easy to understand.

```python

_tf_mapping = None
_pytorch_mapping = None

def get_mapping():
    global _tf_mapping
    global _pytorch_mapping

    if _tf_mapping is None:
        _tf_mapping = xxx

    if _pytorch_mapping is None:
        _pytorch_mapping = xxx
    return _tf_mapping, _pytorch_mapping
```

`unpatch` then just use `_tf_mapping` and `_pytorch_mapping`

Reply:
Or just adding some comments. I am fine with either way.

Reply:
> Does the following code work? Seems more easy to understand.
> 
> ```python
> _tf_mapping = None
> _pytorch_mapping = None
> 
> def get_mapping():
>     global _tf_mapping
>     global _pytorch_mapping
> 
>     if _tf_mapping is None:
>         _tf_mapping = xxx
> 
>     if _pytorch_mapping is None:
>         _pytorch_mapping = xxx
>     return _tf_mapping, _pytorch_mapping
> ```
> 
> `unpatch` then just use `_tf_mapping` and `_pytorch_mapping`

Yeah, that should be much clearer (if works), I will try it later. Thx!

Reply:
it works, changed to global var